include/ensmallen_bits/pso/pso.hpp

```diff
@@ -0,0 +1,138 @@
+/**
+ * @file pso.hpp
+ * @author Chintan Soni
+ * @author Suryoday Basak
+ *
+ * Particle swarm optimization.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_PSO_PSO_HPP
+#define ENSMALLEN_PSO_PSO_HPP
+
+#include "update_policies/lbest_update.hpp"
+#include "init_policies/default_init.hpp"
+#include "init_policies/constrained_init.hpp"
+#include <iostream>
+
+namespace ens {
+
+/**
+ * EXTREMELY DETAILED DESCRIPTION OF THE WORKING OF PSO.
```

Comment:
Would be great to have ``EXTREMELY DETAILED DESCRIPTION``.

Reply:
I will do that, yes!

Reply:
I've done this.